server/api/tests.js

```diff
@@ -7,9 +7,46 @@ const testFileName = 'helloWorld.test.js';
 
 router.post('/', async (req, res, next) => {
   try {
-    const { code } = req.body;
-    fs.writeFileSync('submitted.js', code);
-    res.send(code);
+    // const { code } = req.body;
+    // fs.writeFileSync('submitted.js', code);
+
+    // const { spawn } = require('child_process');
+    // const child = spawn('node', ['submitted.js'], { shell: true });
+    // child.stdout.on('data', (data) => {
+    //   console.log(`stdout: ${data}`);
+    //   res.send(data);
+    // });
+
+    // child.stderr.on('data', (data) => {
+    //   console.error(`stderr: ${data}`);
+    //   res.send(data);
+    // });
+
+    // child.on('close', (code) => {
+    //   console.log(`child process exited with code ${code}`);
+    //   res.send(code);
+    // });
+
+    const objectToHoldStdout = {
+      stdout: '',
+    };
+
+    // TODO: Run a npm test command with a child process
+    const { spawn } = require('child_process');
+    const child = spawn('npm', ['test'], { shell: true });
+    child.stdout.on('data', (data) => {
+      console.log(`stdout: ${data}`);
+      objectToHoldStdout.stdout += data;
+      res.send(objectToHoldStdout.stdout);
+    });
+
+    child.stderr.on('data', (data) => {
+      console.error(`stderr: ${data}`);
+      objectToHoldStdout.stdout += data;
+      res.send(objectToHoldStdout.stdout);
+    });

```

Comment:
Jest output to frontend

Reply:
Gonna try to see if I can do something like this example since this works:
https://stackoverflow.com/a/56095793
```
const util = require('util');
const exec = util.promisify(require('child_process').exec);

async function lsExample() {
  try {
    const { stdout, stderr } = await exec('ls');
    console.log('stdout:', stdout);
    console.log('stderr:', stderr);
  } catch (e) {
    console.error(e); // should contain code (exit code) and signal (that caused the termination).
  }
}
lsExample()
```

Reply:
<img width="378" alt="Screen Shot 2022-09-21 at 7 31 17 PM" src="https://user-images.githubusercontent.com/4698973/191629383-2ef66c4f-262e-47ad-911e-485650db3221.png">